### PR TITLE
General improvements

### PR DESCRIPTION
**Pull request by cabelitos**
The check style will fail in sections that I did not touch.

### CODE REVIEW

File: src/shared/sol-util.c [hunk 1]
```diff
@@ -1026,3 +1026,40 @@ sol_util_unescape_quotes(const struct sol_str_slice slice,
...
+
+        used = strftime_l(sol_buffer_at_end(buf),
+            buf->capacity - buf->used, format, timeptr, c_locale);
+        buf->used += used;
+        return used;
+    }
+#endif
+
+    /**
+       Even with SOL_ATTR_STRFTIME() GCC still wrans that the format parameter was net checked.
```
Comment (edersondisouza) [inline src/shared/sol-util.c]: s/wrans/warns
s/net/not

Comment (cabelitos) [inline src/shared/sol-util.c]: Damn typos. :/

File: src/modules/flow/persistence/persistence.json [hunk 2]
```diff
@@ -323,6 +323,67 @@
...
+      "methods": {
+        "close": "persist_close",
+        "open": "persist_rgb_open"
+      },
+      "name": "persistence/rgb",
+      "options": {
+        "members": [
+          {
+            "data_type": "string",
+            "description": "Name to persist a string packet. If 'storage' is 'fs', name can be a fully fledged path to the file where the value will be stored.",
```
Comment (edersondisouza) [inline src/modules/flow/persistence/persistence.json]: String packet?

File: src/test-fbp/persistence-fs-1.fbp [hunk 3]
```diff
@@ -20,7 +20,7 @@
 # drange_def rgb
```
Comment (edersondisouza) [inline src/test-fbp/persistence-fs-1.fbp]: Maybe update this comment about new files

**Comment by edersondisouza** [comment]
Asides minor comments, persistence related commits looks ok

File: src/shared/include/sol-util.h [hunk 4]
```diff
@@ -1038,6 +1038,23 @@ sol_util_le64_to_cpu(uint64_t val)
  */
 int sol_util_unescape_quotes(const struct sol_str_slice slice, struct sol_buffer *buf);
 
+
+/**
+ * @brief Wrapper around strftime()/strftime_l()
+ *
+ * This is a simple wrraper around strftime()/strftime_l() functions. The
```
Comment (glima) [inline src/shared/include/sol-util.h]: wrapper

File: src/shared/include/sol-util.h [hunk 5]
```diff
@@ -1038,6 +1038,23 @@ sol_util_le64_to_cpu(uint64_t val)
...
+ * @c use_locale parameter is only considered if strftime_l() and newlocale() is
```
Comment (glima) [inline src/shared/include/sol-util.h]: are available

File: src/shared/include/sol-util.h [hunk 6]
```diff
@@ -1038,6 +1038,23 @@ sol_util_le64_to_cpu(uint64_t val)
...
+ * available, otherwise this wrapper will fallback to strftime() - Thus current system's
+ * locale will be considered when formatting the time.
+ *
+ * @param buf The buffer to append the formated time - It must be already initialzed.
```
Comment (glima) [inline src/shared/include/sol-util.h]: the formatted time to

File: src/shared/include/sol-util.h [hunk 7]
```diff
@@ -1038,6 +1038,23 @@ sol_util_le64_to_cpu(uint64_t val)
...
+ * @param format The date format - check strftime man page for accepted formats.
+ * @param timeptr The broken down time struct.
+ * @param use_locale true to use current system locale or false to do not use system's locale.
+ * @return The number of bytes writte, negative errno or 0 on error. (see strftime man page).
```
Comment (glima) [inline src/shared/include/sol-util.h]: written

File: src/modules/flow/platform/platform.c [hunk 8]
```diff
@@ -155,10 +165,34 @@ platform_service_stop_process(struct sol_flow_node *node, void *data, uint16_t p
...
+
+    r = sol_flow_packet_get_string(packet, &name);
+    SOL_INT_CHECK(r, < 0, r);
+    if (mdata->service_name) {
+        sol_platform_del_service_monitor(on_service_state_changed,
+            mdata->service_name, mdata);
+    }
+    r = sol_util_replace_str_if_changed(&mdata->service_name, name);
+    SOL_INT_CHECK(r, < 0, r);
+    return 0;
```
Comment (glima) [inline src/modules/flow/platform/platform.c]: Shouldn't you re-add a new service monitor?

File: src/lib/common/sol-platform-impl-systemd.c [hunk 9]
```diff
@@ -147,16 +150,12 @@ _bus_initialized(sd_bus *bus)
...
 
-    return;
+    return 0;
 
-fail_call:
-    sd_bus_message_unref(m);
-fail_map_properties:
-fail_new_method:
-fail_new_client:
-    sol_bus_close();
```
Comment (glima) [inline src/lib/common/sol-platform-impl-systemd.c]: Don't we need this anymore? Where do we close it?

Comment (cabelitos) [inline src/lib/common/sol-platform-impl-systemd.c]: We close during platform shutdown.

File: src/modules/flow/platform/platform.c (hunk 8 above)
Comment (cabelitos) [inline src/modules/flow/platform/platform.c]: Indeed. I will also need to to send a boolean packet with its state.

File: src/modules/flow/persistence/persistence.json (hunk 2 above)
Comment (cabelitos) [inline src/modules/flow/persistence/persistence.json]: sry

**Comment by cabelitos** [comment]
#1642

**Pull request closed without merging**